server/db/usuarios.js

```diff
@@ -0,0 +1,95 @@
+const Joi = require('joi')
+const db = require('./connection');
+
+const schema = Joi.object().keys({
+	email : Joi.string().required().email({ minDomainAtoms: 2 }).error(new Error('Email invalido.')),
+	nome : Joi.string().min(3).max(75).required().error(new Error('O nome deve conter entre 3 e 75 caracteres.')),
+	senha : Joi.string().required().min(3).max(30).error(new Error('A senha deve conter entre 3 e 30 caracteres.')),
+	curso : Joi.string().required().min(3).max(50).error(new Error('O curso deve conter entre 3 e 50 caracteres.'))
+});
+
+const schemaUpdate = Joi.object().keys({
+	email : Joi.string().email({ minDomainAtoms: 2 }).error(new Error('Email invalido.')),
+	nome : Joi.string().min(3).max(75).error(new Error('O nome deve conter entre 3 e 75 caracteres.')),
+	senha : Joi.string().min(3).max(30).error(new Error('A senha deve conter entre 3 e 30 caracteres.')),
+	curso : Joi.string().min(3).max(50).error(new Error('O curso deve conter entre 3 e 50 caracteres.'))
+});
+
+const tipoUsuarioEnum = {
+	comum: 0,
+	admin: 1
+}
+
+const usuarios = db.collection('usuarios');
+
+function search(req) {
+	if (req.query && req.query['id']) {
+		return getOne(req.query['id']);
+	} 
+	else {
+		return getAll();
+	}
+}
+
+function getAll() {
+	return usuarios.find({},
+						 {"senha": 0 });	
+}
+
+function getOne(id) {
+	return usuarios.findOne({"_id": id},
+                  			{"senha": 0 });;	
+}
+
+function create(usuario) {
+	const resultado = Joi.validate(usuario, schema);
+	if (!resultado.error) {
+		agora = new Date();
+		usuario.dataCriacao = agora;
+		usuario.ultimoAcesso = agora;
+		usuario.tipoUsuario = tipoUsuarioEnum.comum;
+		return usuarios.insert(usuario);
+	} else {
+		const erro = {
+			erro: true,
+			mensagem: resultado.error.message,
+		}
+		return Promise.reject(erro);
+	}
+} 
+
+async function update(usuario) {
+	if (!usuario['id']){
+		throw "Não pode deletar sem um id";
+	}
+	id = usuario["id"];
+	delete usuario["id"];
+	const resultado = Joi.validate(usuario, schemaUpdate);
+	if (!resultado.error) {
+		return usuarios.update({"_id": id}, { $set: usuario});
+	} else {
+		const erro = {
+			erro: true,
+			mensagem: resultado.error.message,
+		}
+		return Promise.reject(erro);
+	}
+}
+
+function remove(req) {
+	if (!req.query['id']){
```

Comment:
Isso é mais um dúvida mesmo: Em geral, se usa o método delete com `query` e não `body`, né? Pq se for, percebi que vagas e reviews estão usando `body` e precisaria mudar

Reply:
Pelo que achei de documentação, é permitido o uso de `body` para `DELETE`, mas não explicitamente recomendado ou desencorajado. 
[Documentação.](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html)
[Stackoverflow sobre body no DELETE.](https://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request)